### PR TITLE
Improve logger signature error handling

### PR DESCRIPTION
**Pull request by instabledesign**
### Signature change
No more context mandatory
No more return error (use errorHandler instead)
- LogInterface
- LoggerInterface

Field now contain his name as attribute

We replace 
`formatter.EnableColor` by `formatter.WithColor`
`formatter.DisplayContext` by `formatter.WithContext`

### CODE REVIEW

File: logger_test.go [hunk 1]
```diff
@@ -4,61 +4,35 @@ import (
...
+	"log"
 	"log/syslog"
 	"os"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/gol4ng/logger/mocks"
+	testing2 "github.com/gol4ng/logger/testing"
 	"github.com/stretchr/testify/assert"
```
Comment (rmasclef) [inline logger_test.go]: I think you can use `testing` as name here now don't you think ?

**Comment by instabledesign** [comment]
>I've understand recently that interfaces should be defined by the consumer of the implementation.

I dont think the consumer must reverse engineer the interface from lib implementation

>That implies that the consumer needs to define its own logger interface and therefore create an adapter that uses our implementation

The provided interface allow you to create another facade/adapter or you can use it directly.
Interface will only add flexibility on the consumer side.

Personally i'll don't want to use a logger that force me to create a facade/adapter.

**Comment by rmasclef** [comment]
> > I've understand recently that interfaces should be defined by the consumer of the implementation.
> 
> I dont think the consumer must reverse engineer the interface from lib implementation

That's what we do when we try to be as less coupled as we can from the outter dependencies 👍 
But I get the point though.

> > That implies that the consumer needs to define its own logger interface and therefore create an adapter that uses our implementation
> 
> The provided interface allow you to create another facade/adapter or you can use it directly.
> Interface will only add flexibility on the consumer side.
> 
> Personally i'll don't want to use a logger that force me to create a facade/adapter.

Technically it does not force you to create a facade, you can use the strong `logger` type if you want in your app ...

I know you don't really like this lib but if we take a look at `zap`, it does not provide any interface for BC break purposes 👍 (see https://github.com/uber-go/zap/issues/381) and therefore let the consumer chose if he wants to use the concrete type directly or create his own logger interface and a zapAdapter in order not to be coupled with zap at all 👍

**Though I'm OK to keep our own interface, I suggest we **_strongly encourage_** users to create their own interface and adapter.**

Is it a good compromise for you ?

cf: https://github.com/uber-go/zap/issues/381

File: README.md [hunk 2]
```diff
@@ -47,22 +47,22 @@ This library expose some quite simple interfaces.
 Simplest one
 ```go
 type LogInterface interface {
-	Log(message string, level Level, context *Context) error
+	Log(message string, level Level, field ...Field)
 }
 ```
 
 The friendly one
```
Comment (rmasclef) [inline README.md]: here I suggest we add a message that strongly encourage the user/consumer to create its own interface and adapter upon this interface 👍

**Pull request merged**